types/pg/index.d.ts

```diff
@@ -9,10 +9,9 @@ import events = require("events");
 import stream = require("stream");
 import pgTypes = require("pg-types");
 
-// tslint:disable-next-line unified-signatures
-export function connect(connection: string, callback: (err: Error, client: Client, done: (err?: any) => void) => void): void;
-// tslint:disable-next-line unified-signatures
-export function connect(config: ClientConfig, callback: (err: Error, client: Client, done: (err?: any) => void) => void): void;
+export function connect(
+    connectionOrConfig: string | ClientConfig,
```

Comment:
When giving a type name `Configuration`, the hint message in IDE will be `function connect(configuration: Configuration)`. It hides the information that `configuration` is either a connection string or a `Config` object. In previous revision I was intended to leave `connect` function separated as overloads to make IDE hint straightforward.

It's just my two cents, from IDE hint perspective. On the other hand, giving a type name follows DRY principle. I can update the code as you advised, please feel free to let me know.

Another thing: I just read pg document on https://node-postgres.com/, and couldn't find any documentation about usage of `connect(connection: string)`. It looks like that pg has deprecated this usage and only leaves `connect(config: ClientConfig)`. I don't know whether the typing should exposing undocumented usage?